### PR TITLE
opnode: Switch to BlockRefs in state loop

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**
This removes having to separately keep track of L1Origin and L1Base.
This is nice because those name are confusing and separately used
and it means less information to keep track of.

### CODE REVIEW

File: opnode/rollup/driver/step.go [hunk 1]
```diff
@@ -84,18 +89,18 @@ func (d *outputImpl) newBlock(ctx context.Context, l2Finalized eth.BlockID, l2Pa
...
+	ref, err := derive.BlockReferences(payload, &d.Config.Genesis)
+	return ref, batch, err
 }
 
 // DriverStep derives and processes one or more L2 blocks from the given sequencing window of L1 blocks.
 // An incomplete sequencing window will result in an incomplete L2 chain if so.
 //
 // After the step completes it returns the block ID of the last processed L2 block, even if an error occurs.
-func (d *outputImpl) step(ctx context.Context, l2Head eth.BlockID, l2Finalized eth.BlockID, unsafeL2Head eth.BlockID, l1Input []eth.BlockID) (out eth.BlockID, err error) {
+func (d *outputImpl) step(ctx context.Context, l2Head eth.L2BlockRef, l2Finalized eth.BlockID, unsafeL2Head eth.BlockID, l1Input []eth.BlockID) (out eth.L2BlockRef, err error) {
```
Comment (protolambda) [inline opnode/rollup/driver/step.go:103]: Shouldn't we be removing `l1Input` now that the `l2Head` has L1 block ID info? What's the point of changing to `BlockRef` without using the extra info?

File: opnode/rollup/driver/step.go [hunk 2]
```diff
@@ -23,14 +23,15 @@ type outputImpl struct {
 	Config rollup.Config
 }
 
-func (d *outputImpl) newBlock(ctx context.Context, l2Finalized eth.BlockID, l2Parent eth.BlockID, l2Safe eth.BlockID, l1Origin eth.BlockID, includeDeposits bool) (eth.BlockID, *derive.BatchData, error) {
-	d.log.Info("creating new block", "l2Parent", l2Parent, "l1Origin", l1Origin, "includeDeposits", includeDeposits)
+func (d *outputImpl) newBlock(ctx context.Context, l2Finalized eth.BlockID, l2Parent eth.L2BlockRef, l2Safe eth.BlockID, l1Origin eth.BlockID) (eth.L2BlockRef, *derive.BatchData, error) {
```
Comment (protolambda) [inline opnode/rollup/driver/step.go:26]: Same question here as in the other comment

File: opnode/rollup/driver/state.go [hunk 3]
```diff
@@ -36,6 +34,11 @@ type state struct {
 	done chan struct{}
 }
 
+// // shouldRunEpoch returns true if there is a full sequencing window between the L2 Safe Head's L1 Origin and the L1 Head.
+// func (s *state) shouldRunEpoch() bool {
+// 	return s.l1Head.Self.Number-s.l2SafeHead.L1Origin.Number >= s.Config.SeqWindowSize
+// }
+
```
Comment (protolambda) [inline opnode/rollup/driver/state.go:41]: ?

**Review by protolambda — CHANGES_REQUESTED** [review]
confused by some of the id -> ref changes

File: opnode/rollup/driver/state.go (hunk 3 above)
Comment (trianglesphere) [reply opnode/rollup/driver/state.go:41]: Dead code from other changes. Will cut.

File: opnode/rollup/driver/step.go (hunk 1 above)
Comment (trianglesphere) [reply opnode/rollup/driver/step.go:103]: This is for timestamps later.

We could also be checking that `l1Input[0].Parent.Hash == l2Head.L1Origin.Hash` as this function advances full epochs at a time.

**Comment by trianglesphere** [comment]
See https://github.com/ethereum-optimism/optimistic-specs/pull/280#issuecomment-1079369761

**Pull request closed without merging**